cfgov/unprocessed/js/organisms/FormSubmit.js

```diff
@@ -0,0 +1,192 @@
+   */
+  function init() {
+    if ( !atomicHelpers.setInitFlag( _baseElement ) ) {
+      return UNDEFINED;
+    }
+    _cachedFields = _cacheFields();
+    _formElement.addEventListener( 'submit', _onSubmit );
+
+    return this;
+  }
+
+  /**
+   * @param {event} event DOM event
+   * @returns {event} DOM event.
+   */
+  function _onSubmit( event ) {
+    event.preventDefault();
+    var errors = _validateForm();
+    if ( errors ) {
+      _displayNotification( _notification.ERROR, errors );
+    } else {
+      _submitForm();
+    }
+
+    return event;
+  }
+
+  /**
+   * @returns {string|undefined} error message.
+   */
+  function _validateForm() {
+    if ( typeof validator === 'function' ) {
+      return validator( _cachedFields );
+    }
+    return UNDEFINED;
+  }
+
+  /**
+   * Displays notification and scrolls it into view if offscreen
+   * @param {type} type of notification
+   * @param {content} content for notification.
+   */
+  function _displayNotification( type, content ) {
+    _notification.setTypeAndContent( type, content );
+    _notification.show();
+    scroll.scrollIntoView( _notificationElement );
+  }
+
+  /**
+   * Sends form data and displays notification on success / failure.
+   * @param {formData} form data object with field name/value pairs
+   */
+  function _submitForm() {
+    var DONE_CODE = 4;
+    var SUCCESS_CODES = {
+      200: 'ok',
+      201: 'created',
+      202: 'accepted',
+      203: 'non-authoritative info',
+      204: 'no content',
+      205: 'reset content',
+      206: 'partial content'
+    };
+    var message;
+    var state = 'ERROR';
+    var xhr = new XMLHttpRequest();
+    xhr.open( 'POST', _formElement.action );
+    xhr.setRequestHeader( 'Content-type', 'application/x-www-form-urlencoded' );
+    xhr.setRequestHeader( 'X-Requested-With', 'XMLHttpRequest' );
+    xhr.onreadystatechange = function() {
+      if ( xhr.readyState === DONE_CODE ) {
+        if ( xhr.status in SUCCESS_CODES ) {
+          var result;
+          try {
+            var response = JSON.parse( xhr.responseText );
+            result = response.result;
+            message = response.message;
+          } catch( err ) {
+            // ignore lack of response
+          }
+          state = result === 'fail' ? 'ERROR' : 'SUCCESS';
+        }
+        _displayNotification( _notification[state],
+                              message || FORM_MESSAGES[state] );
+      }
+    };
+    xhr.send( _serializeFormData() );
+  }
+
+  /**
+   * @returns {obj} form fields, keyed by name.
+   *   Checkboxes and radio fields are stored in array.
+   */
+  function _cacheFields() {
+    var cachedFields = {};
+    var fields = ( _formElement || {} ).elements;
+    var f = 0;
+    var len = fields.length;
+    for ( f; f < len; f++ ) {
+      var field = fields[f];
+      if ( field.name && !field.disabled &&
+         [ 'file', 'reset', 'submit', 'button' ].indexOf( field.type ) === -1 ) {
```

Comment:
Might make sense to store `[ 'file', 'reset', 'submit', 'button' ]` as a variable